### PR TITLE
Experimental benchmarking pipeline

### PR DESCRIPTION
**Pull request by olgavrou**
This pipeline is meant to perform benchmarks between pull request branches and master using google benchmarks. For now it will just print the comparison in the pipeline. We can decide how to report large differences as a second step.

This is pulling in some of the benchmarks created during the flatbuffer benchmarking, so pulling in those related to text and cache parsing.

Once this is merged to master need to update the first step to actually checkout `master`

### CODE REVIEW

File: .scripts/linux/compare-benchmarks.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -e
+set -x
+
+SCRIPT_DIR="$(cd "$(dirname "$0")" && pwd)"
+REPO_DIR=$SCRIPT_DIR/../../
+cd $REPO_DIR
+
+benchmark/tools/compare.py benchmarks "$1" "$2"
```
Comment (jackgerrits) [inline .scripts/linux/compare-benchmarks.sh:9]: Looks like this file is missing from the pr?

Comment (olgavrou) [reply .scripts/linux/compare-benchmarks.sh:9]: compare.py? it's part of the google benchmarks repo which is checked out

Comment (jackgerrits) [reply .scripts/linux/compare-benchmarks.sh:9]: Oh I see

**Review by jackgerrits — APPROVED** [review]

File: .github/workflows/run_benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    container:
+      image: vowpalwabbit/ubuntu1604-build:latest
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+        # with:
+          # ref: master # master doesn't have benchmark scripts right now, change once master is merged
+      - name: Install google benchmarks
+        shell: bash
+        run: ./.scripts/linux/install-benchmark.sh # repo dir
```
[the remark this answers is not in the document]
Comment (olgavrou) [reply .github/workflows/run_benchmarks.yml:18]: I'll remove that it was for me to remember where it is checking out the benchmarks

**Pull request merged**